lamindb/models.py

```diff
@@ -2791,7 +2791,7 @@ def from_dir(
 
     def replace(
         self,
-        data: UPathStr,
+        data: UPathStr | pd.DataFrame | AnnData | MuData,
```

Comment:
I find it not very intuitive that we allow all these other types here, but I guess this is legacy. Hm.

I'd say we should deprecate this, but let's discuss.